xml/System.Diagnostics/DiagnosticSource.xml

```diff
@@ -98,11 +98,11 @@
         <Parameter Name="arg2" Type="System.Object" Index="2" FrameworkAlternate="netcore-2.0;netcore-2.1;netcore-2.2;netcore-3.0" />
       </Parameters>
       <Docs>
-        <param name="name">To be added.</param>
-        <param name="arg1">To be added.</param>
-        <param name="arg2">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="name">The name of the event being written.</param>
+        <param name="arg1">An object that represents the additional context for IsEnabled. Consumers should expect to receive <see langword="null" /> which may indicate that producer called pure IsEnabled(string) to check if consumer wants to get notifications for such events at all. Based on that, producer may call IsEnabled(string, object, object) again with non-<see langword="null" /> context.</param>
+        <param name="arg2">Optional. An object that represents the additional context for IsEnabled. <see langword="null" /> by default. Consumers should expect to receive <see langword="null" /> which may indicate that producer called pure IsEnabled(string) or producer passed all necessary context in <paramref name="arg1" />.</param>
+        <summary>Optional: if there is expensive setup for the notification, you can call IsEnabled before doing this setup with context.</summary>
+        <returns><see langword="true" /> if it's enabled, <see langword="false" /> otherwise.</returns>
```

Comment:
@rpetrusha I agree, so I suggest this modification:

```suggestion
        <returns><see langword="true" /> if the notification event is enabled, <see langword="false" /> otherwise.</returns>
```